org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/DerivedResourcesUpdater.java

```diff
@@ -139,4 +157,37 @@ static void update(IProject workspaceProject, OmniEclipseProject project, Persis
         new DerivedResourcesUpdater(workspaceProject, project).update(persistentModel, monitor);
     }
 
+    /**
+     * Helper class to collect derived resources.
+     */
+    private static final class DerivedResourcesInfo {
+
+        private final IPath projectBuildDir;
+        private final Collection<IPath> nestedProjectPaths;
+        private final Collection<IPath> nestedProjectBuildDirs;
+
+        public DerivedResourcesInfo(IPath projectBuildDir, Collection<IPath> nestedProjectPaths, Collection<IPath> nestedProjectBuildDirs) {
+            this.projectBuildDir = projectBuildDir;
+            this.nestedProjectPaths = nestedProjectPaths;
+            this.nestedProjectBuildDirs = nestedProjectBuildDirs;
+        }
+
+        public IPath getProjectBuildDir() {
+            return this.projectBuildDir == null ? new Path("build") : this.projectBuildDir;
```

Comment:
What I wanted to get at was removing duplication, not changing the behavior on an unknown build dir (that's another discussion). Currently the default (`new Path("build")`) is duplicated here and on line 137. It should only be in one of the two places.